From 031e89e1732af8786a049279d0818d8035ece40f Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 7 Sep 2023 14:07:48 +0200
Subject: [PATCH 0026/1141] ARM: dts: stm32: set pclk limit on dcmi interface
 on stm32mp157a-ev1

Add the property pclk-max-frequency in the dcmi / ov5640 nodes
in stm32mp157a-ev1 in order to ensure that sensor pixel clock
does not exceed the capabilities of the dcmi.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Change-Id: I63ef20b7b3877cca4e7026d7ee79b19ae80e464c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/327621
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/stm32mp157a-ev1.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157a-ev1.dts b/arch/arm/boot/dts/stm32mp157a-ev1.dts
index a4dcf9f21abf..4564d837f324 100644
--- a/arch/arm/boot/dts/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-ev1.dts
@@ -197,6 +197,7 @@ dcmi_0: endpoint {
 			hsync-active = <0>;
 			vsync-active = <0>;
 			pclk-sample = <1>;
+			pclk-max-frequency = <77000000>;
 		};
 	};
 };
@@ -456,6 +457,7 @@ ov5640_0: endpoint {
 				hsync-active = <0>;
 				vsync-active = <0>;
 				pclk-sample = <1>;
+				pclk-max-frequency = <77000000>;
 			};
 		};
 	};
-- 
2.39.2

